server/service/osquery_utils/queries.go

```diff
@@ -1073,6 +1081,52 @@ FROM chrome_extensions`,
 	DirectIngestFunc: directIngestSoftware,
 }
 
+// SoftwareOverrideQueries are used to override software detail query results.  These DetailQueries
+// must include a `SoftwareOverrideMatch` function that returns true if the software row should be
+// overridden with the results of `Query`.
+// Software queries expect specific columns to be present.  Reference the
+// software_{macos|windows|linux} queries for the expected columns.
+var SoftwareOverrideQueries = map[string]DetailQuery{
+	// macos_firefox Differentiates between Firefox and Firefox ESR by checking the RemotingName value in the
+	// application.ini file. If the RemotingName is 'firefox-esr', the name is set to 'Firefox ESR.app'.
+	"macos_firefox": {
+		Query: `
+			WITH app_paths AS (
+				SELECT path 
+				FROM apps 
+				WHERE bundle_identifier = 'org.mozilla.firefox'
+			),	
+			remoting_name AS (
+				SELECT value, path 
+				FROM parse_ini 
```

Comment:
We'll also need `SELECT 1 FROM osquery_registry WHERE active = true AND registry = 'table' AND name = 'parse_ini';` in the discovery query as this is a fleetd (not osquery core) table.

Reply:
And documentation that Firefox ESR is only detected (distinguished from normal Firefox) when using fleetd.

Reply:
great catch!  I added a `generateSQLForAllExists` pattern so we can append multiple `SELECT 1` queries.  (could be used for vscode as well)

Reply:
I also made some additions to the doc generator to enter in the fleetd requirement and a description of override queries.